components/base/wheeled/sensorbase.go

```diff
@@ -0,0 +1,327 @@
+	errCounter := 0
+
+	s.workers.Add(1)
+	utils.ManagedGo(func() {
+		ticker := time.NewTicker(yawPollTime)
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			default:
+			}
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			case <-ticker.C:
+				// imu readings are limited from 0 -> 360
+				currYaw, err := getCurrentYaw(ctx, s.orientation)
+				if err != nil {
+					errCounter++
+					if errCounter > 100 {
+						s.logger.Error(errors.Wrap(
+							err, "imu sensor unreachable, 100 error counts when trying to read yaw angle"))
+						return
+					}
+					return
+				}
+				errCounter = 0 // reset reading error count to zero if we are successfully reading again
+
+				atTarget, overShot, minTravel := getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errBound)
+
+				// if the imu yaw reading is close to 360, we are nearing a full turn,
+				// so we adjust the current reading by 360 * the number of turns we've done
+				if atTarget && (fullTurns > 0) {
+					turnCount++
+					overShot = false
+					minTravel = true
+				}
+
+				if sensorDebug {
+					s.logger.Debugf("minTravel %t, atTarget %t, overshot %t", minTravel, atTarget, overShot)
+					s.logger.Debugf("angleDeg %.2f, increment %.2f, turnCount %d, fullTurns %d",
+						angleDeg, increment, turnCount, fullTurns)
+					s.logger.Debugf("currYaw %.2f, startYaw %.2f, targetYaw %.2f", currYaw, startYaw, targetYaw)
+				}
+
+				// poll the sensor for the current error in angle
+				// check if we've overshot our target by the errTarget value
+				// check if we've travelled at all
+				if fullTurns == 0 {
+					if (atTarget && minTravel) || (overShot && minTravel) {
+						ticker.Stop()
+						if err := s.Stop(ctx, nil); err != nil {
+							return
+						}
+						if sensorDebug {
+							s.logger.Debugf(
+								"stopping base with errAngle:%.2f, overshot? %t",
+								math.Abs(targetYaw-currYaw), overShot)
+						}
+					}
+				} else {
+					if (turnCount >= fullTurns) && atTarget {
+						ticker.Stop()
+						if err := s.Stop(ctx, nil); err != nil {
+							return
+						}
+						if sensorDebug {
+							s.logger.Debugf(
+								"stopping base with errAngle:%.2f, overshot? %t, fullTurns %d, turnCount %d",
+								math.Abs(targetYaw-currYaw), overShot, fullTurns, turnCount)
+						}
+					}
+				}
+			}
+		}
+	}, s.workers.Done)
+	return nil
+}
+
+func getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errorBound float64) (atTarget, overShot, minTravel bool) {
+	atTarget = math.Abs(targetYaw-currYaw) < errorBound
+	overShot = hasOverShot(currYaw, startYaw, targetYaw, dir)
+	minTravel = math.Abs(currYaw-startYaw) > math.Abs(angleDeg*increment)
+	return atTarget, overShot, minTravel
+}
+
+func getCurrentYaw(ctx context.Context, ms movementsensor.MovementSensor,
+) (float64, error) {
+	ctx, done := context.WithCancel(ctx)
+	defer done()
+	orientation, err := ms.Orientation(ctx, nil)
+	if err != nil {
+		return 0, errors.Wrap(
```

Comment:
nit: getCurrentYah shouldn't know the impact of it returning an error. The error message that spin will ignore sensor feedback should be generated by whatever calls it. 
for example, you have 100x retry logic above....so this error is actually not accurate.